File: src/frontend/src/store/modules/mower.js
// initial state
const state = {
  platform: {
    hostname: null,
    operating_system: null,
    platform: null,
    load_average: {
      load1: null,
      load5: null,
      load15: null
    },
    cpu: {
      count: null,
      total: null,
      core_1: null,
      core_2: null,
      core_3: null,
      core_4: null,
      core_5: null,
      core_6: null,
      core_7: null,
      core_8: null,
    },
    memory: {
      total: null,
      available: null
    },
    disk: {
      total: null,
      free: null
    }
  },
  
  battery: {
    status: null,
    voltage_nominal: null,
    voltage_warn: null,
    voltage: null,
    current: null
  },
  
  compass: {
    status: null,
    bearing: null
  },
  
  gps: {
    status: null,
    coordinates: null
  },
  
  
  drive: {
    speed: 100,
    direction: null
  },
  
  cutter: {
    speed: 0
  },
}

// getters
const getters = {}

// actions
const actions = {}

// mutations
const mutations = {
  setMowerDriveSpeed(state, value) {
    state.drive.speed = value
  },
  setDirection(state, value) {
    state.drive.direction = value
  },
  setMowerCutterSpeed(state, value) {
    state.cutter.speed = value
  },
  setMowerState(state, event) {
    state.platform = event.platform
    state.battery = event.battery
    state.compass = event.compass
    state.gps = event.gps
    state.drive = event.drive
    state.cutter = event.cutter
    
    console.log(event)
  }
}

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
}

File: src/control/mower.go
package control

import ()

type MowerStateStruct struct {
	Platform struct {
		Hostname        string `json:"hostname"`
		OperatingSystem string `json:"operating_system"`
		Platform        string `json:"platform"`
		LoadAverage     struct {
			Load1  float64 `json:"load1"`
			Load5  float64 `json:"load5"`
			Load15 float64 `json:"load15"`
		} `json:"load_average"`
		CPULoad struct {
			Count int     `json:"count"`
			Total float64 `json:"total"`
			Core1 float64 `json:"core_1"`
			Core2 float64 `json:"core_2"`
			Core3 float64 `json:"core_3"`
			Core4 float64 `json:"core_4"`
			Core5 float64 `json:"core_5"`
			Core6 float64 `json:"core_6"`
			Core7 float64 `json:"core_7"`
			Core8 float64 `json:"core_8"`
		} `json:"cpu"`
		MemoryUsage struct {
			Total     uint64 `json:"total"`
			Available uint64 `json:"available"`
		} `json:"memory"`
		DiskUsage struct {
			Total uint64 `json:"total"`
			Free  uint64 `json:"free"`
		} `json:"disk"`
	} `json:"platform"`
	Battery struct {
		Status         string  `json:"status"`
		VoltageNominal float64 `json:"voltage_nominal"`
		VoltageWarn    float64 `json:"voltage_warn"`
		Voltage        float64 `json:"voltage"`
		Current        float64 `json:"current"`
	} `json:"battery"`
	Compass struct {
		Status  string `json:"status"`
		Bearing string `json:"bearing"`
	} `json:"compass"`
	GPS struct {
		Status      string `json:"status"`
		Coordinates string `json:"coordinates"`
	} `json:"gps"`
	Drive struct {
		Speed     int    `json:"speed"`
		Direction string `json:"direction"`
	} `json:"drive"`
	Cutter struct {
		Speed int `json:"speed"`
	} `json:"cutter"`
}

var (
	MowerState *MowerStateStruct
)
